Lib/test/test_httpservers.py

```diff
@@ -548,6 +549,80 @@ def test_html_escape_filename(self):
         html_text = '>%s<' % html.escape(filename, quote=False)
         self.assertIn(html_text.encode(enc), body)
 
+class HTTPCompressionTestCase(BaseTestCase):
+    class request_handler(NoLogRequestHandler, SimpleHTTPRequestHandler):
+        pass
+
+    def setUp(self):
+        BaseTestCase.setUp(self)
+        self.cwd = os.getcwd()
+        basetempdir = tempfile.gettempdir()
+        os.chdir(basetempdir)
+        self.data = 10 * b'We are the knights who say Ni!'
```

Comment:
@matrixise 
Are you ok with the changes made since the first version, or do you see other things to do before the PR can be merged ?
If you think a separate PR is better for mimetypes I will submit it, it's just that I prefered to avoid the overhead of reviewing 2 PRs instead of one.